specification/containerservice/Fleet.Management/clusterresourceplacement.tsp

```diff
@@ -0,0 +1,193 @@
+using TypeSpec.Versioning;
+
+namespace Microsoft.ContainerService;
+
+@added(Versions.v2025_03_02_naar)
+@doc("ClusterResourcePlacement is used to select cluster scoped resources, including built-in resources and custom resources, and placement them onto selected member clusters in a fleet.")
+model ClusterResourcePlacement {
+  @doc("PlacementPolicy defines how to select member clusters to place the selected resources. If unspecified, all the joined member clusters are selected.")
+  policy?: PlacementPolicy;
+}
+
+@added(Versions.v2025_03_02_naar)
+@doc("PlacementPolicy contains the rules to select target member clusters to place the selected resources. Only clusters that are both joined and satisfying the rules will be selected. You can only specify at most one of the two fields: ClusterNames and Affinity.")
+model PlacementPolicy {
+  @doc("Type of placement. Default is PickAll.")
+  placementType?: PlacementType;
+
+  @doc("ClusterNames contains a list of names of MemberCluster to place the selected resources. Only valid if the placement type is PickFixed.")
+  clusterNames?: string[];
+
+  @doc("Affinity contains cluster affinity scheduling rules. Defines which member clusters to place the selected resources. Only valid if the placement type is PickAll.") // don't forget to update this when we add PickN.
+  affinity?: Affinity;
+
+  @doc("If specified, the ClusterClusterResourcePlacement's Tolerations. Tolerations cannot be updated or deleted.")
+  @OpenAPI.extension("x-ms-identifiers", [])
+  tolerations?: Toleration[];
+}
+
+@added(Versions.v2025_03_02_naar)
+@doc("PlacementType identifies the type of placement.")
+union PlacementType {
+  @doc("PickAll selects all clusters that match the specified requirements.")
+  PickAll: "PickAll",
+
+  @doc("PickFixed selects a fixed set of clusters")
+  PickFixed: "PickFixed",
+
+  string,
+}
+
+@added(Versions.v2025_03_02_naar)
+@doc("Affinity is a group of cluster affinity scheduling rules.")
+model Affinity {
+  @doc("ClusterAffinity contains cluster affinity scheduling rules for the selected resources.")
+  clusterAffinity?: ClusterAffinity;
+}
+
+@added(Versions.v2025_03_02_naar)
+@doc("ClusterAffinity contains cluster affinity scheduling rules for the selected resources.")
+model ClusterAffinity {
+  @doc("If any affinity requirements specified by this field are not met at scheduling time, the resource will not be scheduled onto the cluster.")
+  requiredDuringSchedulingIgnoredDuringExecution?: ClusterSelector;
+}
+@added(Versions.v2025_03_02_naar)
+@doc("ClusterSelector allows specifying cluster selector terms.")
+model ClusterSelector {
+  @doc("ClusterSelectorTerms is a list of cluster selector terms. The terms are `ORed`.")
+  @OpenAPI.extension("x-ms-identifiers", [])
+  clusterSelectorTerms: ClusterSelectorTerm[];
```

Comment:
some array types are written `foo: Bar[];`, and some are written `foo?: Bar[];` (with a `?`) - is this right?

Reply:
? is to denote optionality of the field.